components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1016,7 +1017,15 @@ export class GitpodServerImpl<Client extends GitpodClient, Server extends Gitpod
             req.setExpose(true);
 
             const client = await this.workspaceManagerClientProvider.get(runningInstance.region);
-            await client.controlPort({ span }, req);
+            try {
+                await client.controlPort({ span }, req);
+            } catch (e) {
+                if (e.code === status.RESOURCE_EXHAUSTED) {
+                    await new Promise((resolve) => setTimeout(resolve, (10*Math.random() + 60)*1000));
```

Comment:
Unfortunately it did not help enough, since we have 5s delay in the supervisor. We will need to implement rate limiting in the supervisor as well.

Reply:
Indeed :(

We should forgo this PR in favour of a more principled approach to rate-limiting the server API surface.